docs/DesignDocs/SGX2-EDMM-driver-interface.md

```diff
@@ -0,0 +1,137 @@
+SGX EDMM Driver Interface Design
+=====================================
+
+## Motivation
+
+This document describes possible driver interfaces to facilitate discussions among SGX runtime implementors (e.g., https://github.com/openenclave/openenclave/pull/3639) on supporting different SGX EDMM flows.
+
+Although interfaces described here are inspired to be as likely as possible a candidate for future kernel adoption, they are not intended to be a proposal for kernel implementation and are assumed to be implemented as an OOT driver. We hope from discussions enabled by this document, requirements and usage models can be identified to help shape future kernel interfaces. 
+
+## Basic EDMM flows
+
+SGX EDMM instructions support dynamic EPC page allocation/deallocation for enclaves and page property modification post-EINIT.  Following are the basic EDMM flows on which other more advanced usages of EDMM can be built.
+
+**Note:** The term "kernel" and "kernel space" are used in this document when general kernel space actions are described whether implemented in an OOT driver or in kernel tree. Kernel specific implementation details will be explicitly stated as "future kernel" or "kernel patches".  And implementation details such as OCalls issued by enclaves, ETRACK and IPI issued in kernel are generally omitted for brevity.
+
+- Allocate a new page at an address in ELRANGE of an enclave. 
+  - This can be an explicit syscall or triggered by #PF  when an unavailable page is accessed.
+  - Kernel issues EAUG for the page. All new pages should have RW permissions initially.
+  - The enclave then issues EACCEPT.
```

Comment:
How can the enclave be sure that 
- the kernel has indeed issues EAUG for the page?
- the a malicious kernel is not able trick the enclave in regards to the confidentiality or permissions of the new page?
- the contents of the page are what it wants to be?

Reply:
When the enclave issues EACCEPT or EACCEPTCOPY, and the instruction succeeds, then the enclave will have an assurance that the page has been EAUG'ed with the correct address and RW permission (EAUG always sets RW in EPCM).